src/libraries/System.Private.CoreLib/src/System/IO/Path.Unix.cs

```diff
@@ -122,13 +122,12 @@ public static bool IsPathRooted(ReadOnlySpan<char> path)
         public static string? GetPathRoot(string? path)
         {
             if (PathInternal.IsEffectivelyEmpty(path)) return null;
-
             return IsPathRooted(path) ? PathInternal.DirectorySeparatorCharAsString : string.Empty;
         }
 
         public static ReadOnlySpan<char> GetPathRoot(ReadOnlySpan<char> path)
         {
-            return PathInternal.IsEffectivelyEmpty(path) && IsPathRooted(path) ? PathInternal.DirectorySeparatorCharAsString.AsSpan() : ReadOnlySpan<char>.Empty;
+            return IsPathRooted(path) ? PathInternal.DirectorySeparatorCharAsString.AsSpan() : ReadOnlySpan<char>.Empty;
```

Comment:
I'm not understanding what this change impacts.

`IsEffectivelyEmpty` is just `path.IsEmpty`, and `IsPathRooted` is just `path.Length > 0 && path[0] == PathInternal.DirectorySeparatorChar` so this condition was just:
```C#
path.IsEmpty && path.Length > 0 && path[0] == PathInternal.DirectorySeparatorChar
```
The first two conditions there are equal.

What does removing the `PathInternal.IsEffectivelyEmpty(path)` check impact, other than removing redundant code?

Am I misreading?

Reply:
Oh, I was misreading, the first two conditions aren't equal, they're polar opposites.  Nevermind :)

Reply:
😃